### ISSUE TITLE
Support configuration to handle more use-cases while not sacrificing the performance

### ISSUE THREAD

**Issue by invidian** [ahmetb/kubectl-aliases#31]
Since the concern with adding more and more options here is the amount of aliases generated and time required for loading them, I think it would be great if such aliases are still in the code, but enabled by default, so user can simply disable things, which he doesn't need.

The configuration would be great, since then user don't have to figure out which parameters are not compatible with what options etc, but simply enables and disables certain features.

As an example, I almost never use JSON output, so I stripped it out from the code, that reduced generation time from 30 seconds to just 5. Also if there are resources, which you never use, user could simple disable them as well.

**Comment by ahmetb** [comment]
Yeah, honestly the combination calculation algorithm I'm using right now is pretty shitty and slow.

If someone can write a nice tool that takes a config file (which we'd have a default checked in to this repo) and once you run it, it outputs an alias file (to stdout), that would be really nice.

That way, people can specify as many resources as they want, run the tool once, and stop bothering the upstream about adding their favorite resource type. :)

**Comment by acramsay** [comment]
I forked and adjusted things to my personal preferences. The latest changes caused python to consume over 2 gigs of RAM and crash. I just solved this by filtering the aliases inside the gen() function rather than after it. The speed is a little improved but more importantly, python now uses only 9k of RAM on my machine so I'm effectively not capped on the inputs anymore. 

For reference, here's the commit that includes the RAM consumption fix. I recommend adding this to the primary fork.
https://github.com/Lodrus/kubectl-aliases/commit/c3918573df96a695a581be757262b69ccf3c1144
